src/Lean/Elab/Binders.lean

```diff
@@ -741,18 +741,12 @@ def elabLetDeclAux (id : Syntax) (binders : Array Syntax) (typeStx : Syntax) (va
       pure (type, val, binders)
   let kind := kindOfBinderName id.getId
   trace[Elab.let.decl] "{id.getId} : {type} := {val}"
-  let result ← if useLetExpr then
-    withLetDecl id.getId (kind := kind) type val fun x => do
+  let result ←
+    withLetDecl id.getId (kind := kind) type val (nondep := !useLetExpr) fun x => do
       addLocalVarInfo id x
```

Comment:
Nice. @kmill No action needed here. I am just marking non trivial changes that I have already checked.